modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SleepTimer.kt

```diff
@@ -31,171 +30,191 @@ class SleepTimer @Inject constructor(
         const val TAG: String = "SleepTimer"
     }
 
-    private var sleepTimeMs: Long? = null
-    private var lastSleepAfterTime: Duration? = null
-    private var lastSleepAfterEndOfChapterTime: Duration? = null
-    private var lastTimeSleepTimeHasFinished: Duration? = null
-    private var lastEpisodeUuidAutomaticEnded: String? = null
+    private var sleepTimerHistory = SleepTimerHistory()
 
-    fun sleepAfter(duration: Duration, onSuccess: () -> Unit) {
-        val sleepAt = System.currentTimeMillis().milliseconds + duration
+    private val _stateFlow: MutableStateFlow<SleepTimerState> = MutableStateFlow(SleepTimerState())
+    val stateFlow: StateFlow<SleepTimerState> = _stateFlow
+
+    fun getState(): SleepTimerState = _stateFlow.value
```

Comment:
```suggestion
    val state: SleepTimerState
        get() = _stateFlow.value
```
Would this be nicer as a variable? If felt strange seeing a `get` method in Kotlin.